lms/djangoapps/support/static/support/jsx/entitlements/components/EntitlementForm/index.jsx

```diff
@@ -0,0 +1,146 @@
+      username: isReissue ? props.entitlement.user : '',
+      comments: 'Add any additional comments here',
+    };
+
+    this.onClose = this.onClose.bind(this);
+    this.handleCourseUUIDChange = this.handleCourseUUIDChange.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handleModeChange = this.handleModeChange.bind(this);
+    this.handleCommentsChange = this.handleCommentsChange.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  onClose() {
+    this.props.closeForm();
+  }
+
+  handleCourseUUIDChange(courseUuid) {
+    this.setState({ courseUuid });
+  }
+
+  handleUsernameChange(username) {
+    this.setState({ username });
+  }
+
+  handleModeChange(mode) {
+    this.setState({ mode });
+  }
+
+  handleCommentsChange(comments) {
+    this.setState({ comments });
+  }
+
+  submitForm() {
+    if (this.state.isReissue) { // if there is an active entitlement we are updating an entitlement
+      const { comments } = this.state;
+      const { entitlement } = this.props;
+      this.props.reissueEntitlement({ entitlement, comments });
+    } else { // if there is no active entitlement we are creating a new entitlement
+      const { courseUuid, username, mode, comments } = this.state;
+      this.props.createEntitlement({ courseUuid, username, mode, comments });
+    }
+  }
+
+  render() {
+    const { isReissue, courseUuid, username, mode, comments } = this.state;
+    const title = isReissue ? 'Re-issue Entitlement' : 'Create Entitlement';
+
+    const body = (
+      <div>
+        <h3> {title} </h3>
+        <InputText
+          disabled={isReissue}
+          name="courseUuid"
+          label="Course UUID"
+          value={courseUuid}
+          onChange={this.handleCourseUUIDChange}
+        />
+        <InputText
+          disabled={isReissue}
+          name="username"
+          label="Username"
+          value={username}
+          onChange={this.handleUsernameChange}
+        />
+        <InputSelect
+          disabled={isReissue}
+          name="mode"
+          label="Mode"
+          value={mode}
+          options={[
+            { label: '--', value: '' },
+            { label: 'Verified', value: 'verified' },
+            { label: 'Professional', value: 'professional' },
+          ]}
+          onChange={this.handleModeChange}
+        />
+        <TextArea
+          name="comments"
+          label="Comments"
+          value={comments}
+          onChange={this.handleCommentsChange}
+        />
+        <div>
+          <Button
+            className={['btn', 'btn-secondary']}
+            label="Close"
+            onClick={this.onClose}
+          />
+          <Button
+            className={['btn', 'btn-primary']}
+            label="Submit"
+            onClick={this.submitForm}
+          />
+        </div>
+      </div>
+    );
+
+    return (
+      <div>
+        { this.props.isOpen ? body : null }
```

Comment:
Could

```javascript
{
  this.props.isOpen
  && body
}
```

work?

Reply:
probably, I'm not sure how much more readable it is? That may just be because I'm pretty comfortable (maybe too comfortable) with ternary operators.